packages/tux-adapter-contentful/src/query-api.ts

```diff
@@ -93,37 +91,32 @@ class QueryApi {
     }
 
     // Add included models to items
-    for (const item of result.items) {
-      this.linkFields(item, linkMap)
+    for (let i = 0; i < result.items.length; i++) {
+      const item = result.items[i]
+      result.items[i] = this.linkEntry(item, linkMap)
     }
   }
 
-  private linkFields(item: any, linkMap: LinkMap) {
-    if (!item) {
-      return
+  private linkEntry(item: any, linkMap: LinkMap): any {
+    const hasSys = !!item.sys
+    const isLink = hasSys && item.sys.type === 'Link'
+    const isArray = item instanceof Array
+
+    if (item.fields) {
+      Object.keys(item.fields).forEach(fieldName => {
+        item.fields[fieldName] = this.linkEntry(item.fields[fieldName], linkMap)
+      })
```

Comment:
I would have performed array handling inside this fields loop, but I see how you're thinking this. 

If anything, I'd change this to an `else if` below. Then it's more clear that the function is handling 3 distinct cases.